### ISSUE TITLE
Allow Wush to run on non root devices

### ISSUE THREAD

**Issue by SerJaimeLannister** [coder/wush#25]
Hello I was thinking of creating a multi platform application on wush (just thinking , I liked magic wormhole but its golang library was a bit rough and I really liked golang and like your project's approach)
On offtopic I was the author to (https://news.ycombinator.com/item?id=38876954) so I have had a keen interest in this space

Since wush uses wireguard and wireguard needs root access , I found that there is this wireguard compliant non root alternative called wiretap (https://github.com/sandialabs/wiretap)which could allow the server to run rootlessly and also wireproxy (https://github.com/pufferffish/wireproxy)
which could connect to clients rootlessly (or maybe wiretap can do that as well)

Anyhow adding this support could be really interesting
I did somewhere hear that tailscale can be configured in a non rootless device as when I was looking at wireproxy and the likes. some people said that something like this could work directly in headscale , so please look into that as well.. 

I hope that you could look into these matters.

**Comment by coadler** [comment]
Hey! `wush` doesn't actually require root. We use wireguard-go, which is entirely user space.

**Comment by SerJaimeLannister** [comment]
I am not exactly sure.
First of all your script requires root to run (maybe to but it in bin path on linux)
but then I just went to the releases page and no worries I installed it and added to a non root vps (which is behind a nas)

and then I did ./wush serve and it gave me an auth key and when I tried to add a random file to , using wush cp if I remember correctly , then it just kept on loading ,saying wireguard configuration or something like that. I am going to update you on it & didn't give any error , but the file wasn't getting pasted.

**Comment by SerJaimeLannister** [comment]
then it just said bringing wireguard up on my device from which I was trying to send file from as a trial (it has root access) 

**On my non rooted device behind nat on which I ran wush serve**
Use this key to authenticate other  wush  commands to this instance.
WireGuard is ready
SSH server enabled
File transfer server enabled
15:11:34 Received connection request over DERP from example@test

and on my device it showed (I have changed some data since I was not sure if that has personal information or not but still)

**On my rooted device from which I was trying to send file it 
showed**
 
 wush cp random.file
Auth information:
        > Server overlay STUN address:  XYZ
        > Server overlay DERP home:     XYZ
        > Server overlay public key:    [XYZ]
        > Server overlay auth key:      [XYZ]
Bringing WireGuard up..

and then it just keeps on loading (> 5 -10 minutes)  and it doesn't show up any errors but its clearly not working)

**Comment by SerJaimeLannister** [comment]
hey any updates regarding this ?

**Comment by coadler** [comment]
Hey @SerJaimeLannister, are you able to run `wush cp` in verbose mode?

`wush cp -v random.file`

This should let us know where it's hanging

**Comment by SerJaimeLannister** [comment]
yes this -v command does absolutely nothing in terms of additional logs.
![image](https://github.com/user-attachments/assets/33b70d3e-a474-46ec-88f0-45a186df1779)
![image](https://github.com/user-attachments/assets/29729cc6-2077-41cb-95c5-3d4fefe97dbb)

**Comment by coadler** [comment]
I may have fixed this in https://github.com/coder/wush/pull/47. Going to push a new release. Let me know if you're still seeing issues with the new version.

**Comment by SerJaimeLannister** [comment]
> Thanks for the additional info. Could you confirm what os/architecture `wush` is stuck on? I'm currently digging into something similar on win11/arm64.

Both the devices are on linux
(also I think x86_64) 

One of the devices is behind a nat

**Comment by SerJaimeLannister** [comment]
Hello I have tried it again on multiple servers behind nats and I  have also installed the latest version and its unfortunately still not working.

**Comment by SerJaimeLannister** [comment]
To me it seems that tailscale servers are blocked on the servers , though this is only a hypothesis

**Comment by SerJaimeLannister** [comment]
Hey any updates , this means rather lot to me
